### PR TITLE
DC-1238: Add dummy genomic data to test OMOP datasets

### PR DESCRIPTION
**Pull request by pshapiro4broad**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/DC-1238

## Addresses

Update test OMOP data for integration / connected tests and for local dataset setup to include a `sample` table with dummy genomic data.

## Summary of changes

- add Sample table to schema and settings for test data
- minor changes to `setup_tdr_resources.py`
- rename a `.jsonl` file to `.json`

## Testing Strategy

- integration / connected tests
- ran `setup_tdr_resources.py` locally

### CODE REVIEW

**Review by s-rubenstein — APPROVED** [review]
Looks reasonable

File: tools/setupResourceScripts/setup_tdr_resources.py [hunk 1]
```diff
@@ -58,38 +58,23 @@ def wait_for_jobs(clients, jobs):
...
-                    ),
-                )
-            else:
-                print(
-                    f"Skipping ingest of {dataset_to_upload['name']}/{table} because it is empty"
-                )
+        ) as table_jsonl:
+            ingest_request = {
+                "format": "array",
+                "records": [json.loads(line) for line in table_jsonl],
```
Comment (fboulnois) [inline tools/setupResourceScripts/setup_tdr_resources.py:69]: Much cleaner! 👍

File: tools/setupResourceScripts/files/OMOPDataset/sample.json [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{"collaborator_participant_id": "77", "collaborator_sample_id": "SM-77", "contamination_rate": 0.001, "exome_gvcf_index_path": "gs://fc-00000000-0000-0000-0000-000000000000/sample.wes.hard-filtered.gvcf.gz.tbi", "exome_gvcf_md5_path": "gs://fc-00000000-0000-0000-0000-000000000000/sample.wes.hard-filtered.gvcf.gz.md5", "exome_gvcf_path": "gs://fc-00000000-0000-0000-0000-000000000000/sample.wes.hard-filtered.gvcf.gz", "genome_crai_path": "gs://fc-00000000-0000-0000-0000-000000000000/sample.cram.crai", "genome_cram_md5_path": "gs://fc-00000000-0000-0000-0000-000000000000/sample.cram.md5", "genome_cram_path": "gs://fc-00000000-0000-0000-0000-000000000000/sample.cram", "mapped_percentage": 99.86, "mean_off_target_coverage": 3.46, "mean_target_coverage": 53.8, "percent_target_bases_at_10x": 98.36, "percent_wgs_bases_at_1x": 94.77, "reblocked_gvcf": "gs://fc-00000000-0000-0000-0000-000000000000/sample.rb.g.vcf.gz", "reblocked_gvcf_index": "gs://fc-00000000-0000-0000-0000-000000000000/sample.rb.g.vcf.gz.tbi", "total_bases": 23772304683}
```
Comment (fboulnois) [inline tools/setupResourceScripts/files/OMOPDataset/sample.json:1]: this looks like a `jsonl`?

**Review by fboulnois — COMMENTED** [review]
See below:

**Review by rushtong — APPROVED** [review]
Looks reasonable to me. My nagging concern here would be the tight coupling between the test classes (integration, service, and DAO tests) to the single dataset file which is how these tests were originally written.

File: tools/setupResourceScripts/files/OMOPDataset/sample.json (hunk 2 above)
Comment (pshapiro4broad) [reply tools/setupResourceScripts/files/OMOPDataset/sample.json:1]: Yes, all the data tables are jsonl. I found a number of changes like this that would be an improvement, but decided to limit the work done in this PR, I'll make another tech debt ticket to further clean up this code

**Comment by pshapiro4broad** [comment]
@rushtong 
> My nagging concern here would be the tight coupling between the test classes (integration, service, and DAO tests) to the single dataset file which is how these tests were originally written.

Is your concern about having multiple tests depend on the same files, or having tests depend on files outside the test itself, or something else?

For tests that require a lot of assets to set up, I think it makes sense to use external data files like this, rather than embedding the json in the test code, or adding objects that define the models for the json data. But I agree that this separation means that you have to be extra careful when updating the data files.

**Review by rjohanek — APPROVED** [review]
this looks good! are the datasets in dev/prod going to be updated as a part of this work? I'm excited to see the new table in action

File: tools/setupResourceScripts/files/OMOPDataset/sample.json (hunk 2 above)
Comment (fboulnois) [reply tools/setupResourceScripts/files/OMOPDataset/sample.json:1]: The file extension is `json`, should it be `jsonl`?

**Comment by rushtong** [comment]
@pshapiro4broad
> Is your concern about having multiple tests depend on the same files, or having tests depend on files outside the test itself, or something else?
> 
> For tests that require a lot of assets to set up, I think it makes sense to use external data files like this, rather than embedding the json in the test code, or adding objects that define the models for the json data. But I agree that this separation means that you have to be extra careful when updating the data files.

My concern is that each of the different test classes are now dependent on the same underlying data such that any change to the underlying data affects multiple tests in potentially different ways - essentially the last point that you make.

**Comment by pshapiro4broad** [comment]
@rushtong 
> My concern is that each of the different test classes are now dependent on the same underlying data such that any change to the underlying data affects multiple tests in potentially different ways - essentially the last point that you make.

I think this can go either way. Having one test data set for a number of tests does mean that changing the data could require many tests to change. On the other hand, creating custom test data for each test would lead to duplicated (and possibly identical) test data; anytime an input schema changes many different files would need to be updated. So far I believe the shared data hasn't been a big issue. Part of the issue here is that the test data isn't a single file but is a collection of files that must all agree in certain ways.

In any event a change like this would be outside the scope of the original ticket, so I can create a backlog ticket if you think this is worth looking into.

File: tools/setupResourceScripts/files/OMOPDataset/sample.json (hunk 2 above)
Comment (pshapiro4broad) [reply tools/setupResourceScripts/files/OMOPDataset/sample.json:1]: I've created https://broadworkbench.atlassian.net/browse/DC-1287 for this follow-up work.

To be clear, I can't rename this file to `jsonl` without changing all the files, and the script, so I think that should be deferred for now.

**Pull request merged**